prow/deck/jobs/jobs.go

```diff
@@ -141,7 +141,50 @@ func (ja *JobAgent) GetJobLog(job, id string) ([]byte, error) {
 	}
 	ja.mut.Unlock()
 	if !ok {
-		return nil, fmt.Errorf("no such job found: %s (id: %s)", job, id)
+		return kube.ProwJob{}, fmt.Errorf("no such job found: %s (id: %s)", job, id)
+	}
+	return j, nil
+}
+
+// GetJobLogTail returns the last n bytes of the job logs, works for both kubernetes and jenkins agent types.
+func (ja *JobAgent) GetJobLogTail(job, id string, n int64) ([]byte, error) {
+	j, err := ja.getProwJob(job, id)
+	if err != nil {
+		logrus.WithError(err).Error("Error getting ProwJob.")
+	}
+	if j.Spec.Agent == kube.KubernetesAgent {
+		client, ok := ja.pkcs[j.ClusterAlias()]
+		if !ok {
+			return nil, fmt.Errorf("cannot get logs for prowjob %q with agent %q: unknown cluster alias %q", j.ObjectMeta.Name, j.Spec.Agent, j.ClusterAlias())
+		}
+		return client.GetLogTail(j.Status.PodName, kube.TestContainerName, n)
+	}
+	for _, agentToTmpl := range ja.c.Config().Deck.ExternalAgentLogs {
+		if agentToTmpl.Agent != string(j.Spec.Agent) {
+			continue
+		}
+		if !agentToTmpl.Selector.Matches(labels.Set(j.ObjectMeta.Labels)) {
+			continue
+		}
+		var b bytes.Buffer
+		if err := agentToTmpl.URLTemplate.Execute(&b, &j); err != nil {
+			return nil, fmt.Errorf("cannot execute URL template for prowjob %q with agent %q: %v", j.ObjectMeta.Name, j.Spec.Agent, err)
+		}
+		resp, err := http.Get(b.String())
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
```

Comment:
Shouldn't you trim the response down to the last n bytes here?

Reply:
s/bytes/lines/

Reply:
the current implementation uses last n bytes, but yes I absolutely should

Reply:
addressed in latest commit